src/libraries/Microsoft.PowerFx.Core/Entities/External/InternalTableParameters.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Entities.Delegation;
+using Microsoft.PowerFx.Core.Entities.QueryOptions;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Functions.Delegation.DelegationMetadata;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.UtilityDataStructures;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    internal class InternalTableParameters : IExternalTabularDataSource
```

Comment:
>InternalTableParameters [](http://example.com/codeflow?start=19&length=23)

It's not clear from the name what this class supposed to do. Can you please add some comments here? Why parameters?

Reply:
Lots of evolutions/updates (...), let's call it DataSource